merge/set_test.go

```diff
@@ -189,11 +189,11 @@ func TestUpdateSet(t *testing.T) {
 			Object: `
 				list:
 				- a
-				- aprime
 				- b
+				- aprime
 				- c
-				- cprime
 				- d
+				- cprime
```

Comment:
I dont understand how this PR results in the ordering of objects in merged arrays changing. Can you describe why this result is correct for this tests's series of Apply/Update ops.

Reply:
Yeah, this test was actually moot because the comparison function wasn't checking for order. So that test started failing as I changed the comparison function.

Reply:
Ah so the behavior did not change but the comparison used by the test did